### ISSUE TITLE
Beta 13 Seems to Hang when running Quick_Refresh Mac OS Big Sur

### ISSUE THREAD

**Issue by oboe45** [openaudible/openaudible#546]
The Quick refresh command runs for hours and hours and as far as I can tell it never completes.  While running the Quick Refresh, It does not seem to be updating the log page, and, I cannot copy/paste from the log page at that time.

The OA log shows an info line 
INFO 12:21:04 org.openaudible.desktop.swt.manager.menu.commandcenter execute
INFO Command: Quick_Refresh.
 Then nothing is on the OA Log

The Audible Library Update Panel Shows "Loaded 908 books" then at mid-panel "InProgress:10 queue 901". It stays like this with no changes for at least two hours,

Using the Mac Activity Monitor, the OA App is using 844.91 % CPU which, 2 GHz Quad-Core Intel Core i5 MacBook Pro - is impossible - my understanding of the % shown is that the processor is therefore 200% overloaded!  Not sure what that CPU% actually means, except that I have never seen the number code to 100%, ever.

I hit the cancel button on the Update dialog and a note appears on the dialog saying cancelling -  it stays like that for at least a couple of hours - no entry is written on the log, and the cancelling process goes on forever - I left it for a very long time, then used the system command in Mac to cancel the app.  Note that during the Cancelling period,, the CPU % still showed over 800% with more than 60 active threads.

What is happening here?  

The generated web page I produced after cancelling the Quick Refresh looks quite good - all of the book details are complete except for the "Download" link for the 15 m4b books in the converted library which are all blank.  

All of the books from the Audible.CA are listed correctly with all details correct excpt those noted in the preceding paragraph,

Also, on the page with "Show Images" selected, the Narrated link is incorrectly pointing to the Author at Audible.ca.   Note that the links on the Non-Image page are correctly linked.

**Comment by openaudible** [comment]
Refresh is supposed to go and fetch any missing cover art. We are supposed to use 10 threads.. and fetching could involve ffmpeg to get the image from the media file (aax,mp3,m4a,m4b) or from the internet if there is a url for the image. I'll reduce the threads to 3 and put up a progress dialog and print additional diagnostics. But it shouldn't peg the cpu. 

Update: Found bad code that caused the cpu problem. Will post a new version ASAP. Thanks for the report!

**Comment by openaudible** [comment]
Fixed in beta 14. Thank you for the great reports.

**Comment by oboe45** [comment]
Installed 2,4,1,4  Beta 14 today,

1. Logged out, then Logged in and check the preferences - all seem OK
2. Panel says I have 910 Books, 910 M4b Files . 0 to
convert/download/downloading and converting
3. The Web page from previous version has following problems:
a No file links to the M4B files are shown in the modal dialog
b. All files except 1 - Discworld Jingo have cover pages shown on the site
- not an issue since the Page on Audible.ca also shows no cover page for
this book, its one of those that Audible says is out of print or
something similar. The book still downloads despite this, but no cover is
ever found.
c.. On the "show Images" page, Narrated Links are present, but all of them
point to the Author page, not to the Narrator page at Audible;  All of the
Author links work correctly.
d. When "Show Images" is off, the links on the page point correctly to the
Author and to the Narrator page.

Next,

4. I Ran a quick library sync - it ran quickly then said "*Your library
appears up to date! Go buy more Audible books! *(If new books weren't
found, check the console.)
5. Panel says I have 910 Books, 910 M4b Files 0 to
convert/download/downloading and converting
6. Ran a full sync -  took a little longer, but produced exactly the same
result as 4 and 5 above.
7. Ran Export Web Page - ran with no issues
8. Following problems still exist on this beta - 2,4,1,4
a No file links to the M4B files are shown in the modal dialog
b. All files except 1 - "Discworld Jingo" have cover pages shown on the site
c.. On the "show Images" page, Narrated Link re present, but all of them
point to the Author page, not to the Narrator page at Audible;  All of the
Author links work correctly.
d. When "Show Images" is off, the links on the page point correctly to the
Author and to the Narrator page.

In other words no changes at all between the previous beta and the current
one except that full library sync now completes in a timely fashion, and it
does not consume more than 20% cpu time at any point.

Is it possible to note in the web page header the Version number of the OA
software used to generate the page, it makes it easier to make sure I am
using the correct version.

Just about the only real issue is the absence of the file links on the
modal dialog page.  Other issues are minor, personally the Narrator link is
not useful, the author link is useful and works correctly.

**Comment by openaudible** [comment]
Awesome report. Will fix ASAP.
